File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/entities/dtos/CreateAccountDto.java
package com.decrux.pocketr.api.entities.dtos;

import java.time.LocalDate;

public record CreateAccountDto(
    String name,
    String type,
    String currency,
    Long openingBalanceMinor,
    LocalDate openingBalanceDate
) {
    public CreateAccountDto {
        RequestDtoValidator.requireNotBlank(name, "name");
        RequestDtoValidator.requireMaxLength(name, 255, "name");
        RequestDtoValidator.requireNotBlank(type, "type");
        RequestDtoValidator.requireCurrencyCode(currency, "currency");
    }

    public String getName() {
        return name;
    }

    public String getType() {
        return type;
    }

    public String getCurrency() {
        return currency;
    }

    public Long getOpeningBalanceMinor() {
        return openingBalanceMinor;
    }

    public LocalDate getOpeningBalanceDate() {
        return openingBalanceDate;
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr/api/entities/dtos/CreateAccountDto.kt
package com.decrux.pocketr.api.entities.dtos

import java.time.LocalDate

data class CreateAccountDto(
    val name: String,
    val type: String,
    val currency: String,
    val openingBalanceMinor: Long? = null,
    val openingBalanceDate: LocalDate? = null,
)
